cachito/web/content_manifest.py

```diff
@@ -232,3 +237,246 @@ def generate_icm(self, image_contents=None):
         icm = deepcopy(BASE_ICM)
         icm["image_contents"] = image_contents or []
         return deep_sort_icm(icm)
+
+
+class ContentManifestPackage:
+    """
+    A package within a content manifest.
+
+    It is used primarily to generate a package URL (purl).
+    """
+
+    def __init__(self, name, type, version, dev=False, dependencies=[], path=None, replaces=None):
+        """
+        Initialize ContentManifestPackage.
+
+        :param str name: the package's name
+        :param str type: the package manager used to process this package. Should be one
+            of the following: gomod, go-package, npm, yarn, pip or git-submodule
+        :param str version: the package's version
+        :param bool dev: if the package is only used in a dev environment
+        :param list[dict] dependencies: the package's dependencies as a list of JSON
+            dicts
+        :param str path: relative path to package from root of repository
+        :param str replaces: the dependency that is being replaced by this one in the
+            context of the request
+        """
+        self.name = name
+        self.type = type
+        self.version = version
+        self.dev = dev
+        self.path = path
+        self.replaces = replaces
+
+        self.dependencies = [
+            ContentManifestPackage.from_json(dependency) for dependency in dependencies
+        ]
+
+    def __repr__(self):
+        return (
+            f"<{self.__class__.__name__} name={self.name}, type={self.type}, "
+            f"version={self.version}>"
+        )
+
+    @property
+    def hash(self):
+        """Generate the package hash based on its name, type and version."""
+        return hash(f"{self.name}{self.type}{self.version}")
```

Comment:
Instead of hashing by the hash, I think we should hash by a unique key, such as a tuple of `(name, version, type, dev)`. Alternatively, you could give this class `__hash__` and `__eq__` to make it properly hashable.

Reply:
Done!